### DISCUSSION TITLE
[Uno.Wasm.Bootstrap] RuntimeError: table index is out of bounds in function pointer call

### DISCUSSION THREAD

**Discussion by Beyley** [unoplatform/uno#10836] [Q&A]
Im unable to get a proper stacktrace as firefox crashes if i try to open the wasm file to view, however, using print i was able to shrink the problem down to these 2 lines of code
![image](https://user-images.githubusercontent.com/34501060/209750803-94f1343f-1557-4370-a512-8c82329f440e.png)
![image](https://user-images.githubusercontent.com/34501060/209750827-c709dc35-49e7-441a-8d07-904922ef6cf8.png)
branch to clone is [wasm-take-3](https://github.com/dotnet/Silk.NET/tree/wasm-take-3) 
and project is [`src/Lab/Experiments/Triangle/TriangleWasm.csproj`](https://github.com/dotnet/Silk.NET/blob/wasm-take-3/src/Lab/Experiments/Triangle/TriangleWasm.csproj)

**Comment by jeromelaban** [comment]
Thanks for the report. Such an error generally means that the code is trying to invoke a method which has not been registered as invocable. It's likely something that WebAssembly does not support altogether (some types of dynamic invocations) or that the .NET runtime does not support properly.

If you can create a smaller sample for the scenario you're trying to build, we may be able to create an issue for the .NET runtime, or create a workaround for that scenario.

**Reply by Beyley** [reply]
Ive been toying with this over the past month trying to reduce the surface area for the error and found very little, all i know is that
[this project works](https://github.com/dotnet/Silk.NET/tree/wasm-take-3/src/Lab/Experiments/BlankWindow)
but [this one doesnt](https://github.com/dotnet/Silk.NET/tree/wasm-take-3/src/Lab/Experiments/Triangle)

ive tried copying all the differences from `BlankWindowWasm.csproj` into `TriangleWasm.csproj` and cant get it to start working
i also tried copying the init code from `BlankWindow` into `Triangle` and it still stays broken, im starting to lean towards it being something in the project file, but i do not know what, they seem functionally identical (i tried matching the versions of uno wasm bootstrap to no avail)

**Reply by jeromelaban** [reply]
This PR https://github.com/unoplatform/Uno.Wasm.Bootstrap/pull/671 (8.0.0-dev.92) may have had an impact on your specific scenario, could you try and see if it helps?

**Reply by Beyley** [reply]
> This PR [unoplatform/Uno.Wasm.Bootstrap#671](https://github.com/unoplatform/Uno.Wasm.Bootstrap/pull/671) (8.0.0-dev.92) may have had an impact on your specific scenario, could you try and see if it helps?

Unfortanately updating to that did not fix the issue

**Reply by Beyley** [reply]
Tried updating to 8.0.0-dev.122, and it unfortanately does not fix the bug, however updating to 8.0.0-dev.132 causes another problem to happen during compliation, tried looking at the binlog and it seems to be related to copying files during linking, but im not 100% sure, binlog is here, just remove the .zip extension since github doesnt let you upload .binlog files
[msbuild.binlog.zip](https://github.com/unoplatform/uno/files/11228981/msbuild.binlog.zip)
same project file as [this](https://github.com/dotnet/Silk.NET/blob/wasm-take-3/src/Lab/Experiments/Triangle/TriangleWasm.csproj) but with updated dependencies to the latest versions, that being 
```
        <PackageReference Include="Uno.Foundation.Runtime.WebAssembly" Version="4.9.0-dev.555" />
        <PackageReference Include="Uno.Wasm.Bootstrap" Version="8.0.0-dev.132" />
        <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="8.0.0-dev.132" PrivateAssets="all" />
```

**Reply by jeromelaban** [reply]
Indeed there's a problem with some files copy, though I'm not able to reproduce it locally. Are you running on Linux? Is there a public branch I could try to reproduce the issue with?
